src/peft/peft_model.py

```diff
@@ -721,16 +748,82 @@ def load_adapter(self, model_id: str, adapter_name: str, is_trainable: bool = Fa
                     no_split_module_classes=no_split_module_classes,
                     low_zero=(device_map == "balanced_low_0"),
                 )
+
             if isinstance(device_map, str):
                 device_map = infer_auto_device_map(
                     self, max_memory=max_memory, no_split_module_classes=no_split_module_classes
                 )
+
+            prefix = "base_model.model."
+            # rename offload index weight and model names
+            target_modules = self.peft_config["default"].target_modules
+            if offload_index:
+                keys = list(offload_index.keys())
+                block_id = keys[0].split(".")[0] + "."  # for writing safetensors key
+
+                for key in keys:
+                    if any(module in key for module in target_modules):
+                        suffix_pos = key.rfind(".")
+                        new_key = prefix + key[:suffix_pos] + ".base_layer" + key[suffix_pos:]
+                    else:
+                        new_key = prefix + key
+                    offload_index[key]["weight_name"] = new_key
+                    offload_index[new_key] = offload_index[key]
+                    del offload_index[key]
+
+                # rename safetensors for dispatch
+                file_seen = set()
+                for new_key in list(offload_index.keys()):
+                    fname = offload_index[new_key]["safetensors_file"]
+
+                    # make a new file name
+                    new_fname_list = list(fname.split("/"))
```

Comment:
Here and below, let's not hard-code `"/"` but use `os.path.sep`.

Reply:
Replaced with `os` functions